### PR TITLE
Close #148 - add unit test for SalesProcessServiceImpl

### LINKED ISSUE #148: Write unit tests for `org.trebol.operation.services.SalesProcessServiceImpl`
From #88 

`SalesProcessServiceImpl ` should have unit tests that ensure it complies with the interface and methods it implements, `ISalesProcessService`.

No need to provide coverage for `private` methods, only the `@Override public` ones from the aforementioned interface.

This interface provides three documented methods.

### PR DESCRIPTION
**Pull request by mepox**
## PR Checklist

- [x] Read the [contribution guidelines](/CONTRIBUTING.md)
- [x] Added a brief description of these changes to the top of the [changelog](/CHANGELOG.md)
- [x] Changes in code meet test criteria (running `mvn test` returns exit code 0, without errors)

## PR Type

- [x] Test-related

## Summary

Add unit test for SalesProcessServiceImpl, issue #148 

- Used @Nested classes to separate the methods
- Mock objects were setup at the beginning of each test methods so it's easier to see what are they (no need to scroll up and down)

Thanks @NyorJa for the help how to mock multiple interfaces which uses generics #167

### CODE REVIEW

File: src/test/java/org/trebol/operation/services/SalesProcessServiceImplTest.java [hunk 1]
```diff
@@ -0,0 +1,454 @@
...
+import org.trebol.jpa.repositories.ISellStatusesJpaRepository;
+import org.trebol.jpa.services.GenericCrudJpaService;
+import org.trebol.jpa.services.ITwoWayConverterJpaService;
+import org.trebol.pojo.ProductPojo;
+import org.trebol.pojo.SellPojo;
+
+@ExtendWith(MockitoExtension.class)
+class SalesProcessServiceImplTest {
+	
+	@InjectMocks	SalesProcessServiceImpl instance;
```
Comment (NyorJa) [inline src/test/java/org/trebol/operation/services/SalesProcessServiceImplTest.java:40]: //	@InjectMocks
  SalesProcessServiceImpl instance;
	@Mock	GenericCrudJpaService<SellPojo, Sell> crudService;
	@Mock	ISalesJpaRepository salesRepository;
	ITwoWayConverterJpaService<SellPojo, Sell> converterService;
	ITwoWayConverterJpaService<ProductPojo, Product> productConverterService;
	@Mock	ISellStatusesJpaRepository sellStatusesRepository;
	
	@Mock	ISellDetailsJpaRepository sellDetailsRepository;	
	//@Mock	ProductsConverterJpaServiceImpl productConverterService;

  @BeforeEach
  public void setUpBeforeEach() {
    converterService = (ITwoWayConverterJpaService<SellPojo, Sell>) Mockito.mock(ITwoWayConverterJpaService.class);
    productConverterService = (ITwoWayConverterJpaService<ProductPojo, Product>) Mockito.mock(ITwoWayConverterJpaService.class);
    instance = new SalesProcessServiceImpl(crudService, salesRepository, sellDetailsRepository, sellStatusesRepository, converterService, productConverterService);
  }

We manual inject the dependencies. That's the solution in handling generics

Comment (NyorJa) [reply src/test/java/org/trebol/operation/services/SalesProcessServiceImplTest.java:40]: yeah. Its kind of tricky but casting the generic dependencies solves the issue

Comment (NyorJa) [reply src/test/java/org/trebol/operation/services/SalesProcessServiceImplTest.java:40]: No worries man. Salud!

**Comment by mepox** [comment]
Hi @bglamadrid ! I added the comments. Also, I found better to rename that "converterService" so it's clear which convertService is which one. 😄

**Review by bglamadrid — APPROVED** [review]

**Pull request merged**